ext/dcos-installer/dcos_installer/action_lib/configure.py

```diff
@@ -6,34 +7,45 @@
 import gen
 import pkgpanda
 import gen.installer.bash
-from dcos_installer.util import SERVE_DIR
+from dcos_installer.util import ARTIFACT_DIR, GENCONF_DIR, SERVE_DIR
 
 log = logging.getLogger(__name__)
 
 
 def do_configure(gen_config):
+    # Generate complete config.
     gen_config.update(get_gen_extra_args())
-
-    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
-
     do_validate_gen_config(gen_config)
-
     gen_out = gen.generate(arguments=gen_config)
-    gen.installer.bash.generate(gen_out, SERVE_DIR)
 
-    # Get bootstrap from artifacts
-    fetch_bootstrap(gen_out.arguments['bootstrap_id'])
-    # Write some package metadata
-    pkgpanda.util.write_json('/genconf/cluster_packages.json', gen_out.cluster_packages)
+    # Generate install/upgrade prereqs.
+    subprocess.check_output(['mkdir', '-p', SERVE_DIR])
+    gen.installer.bash.generate(gen_out, SERVE_DIR)
+    fetch_bootstrap(ARTIFACT_DIR, SERVE_DIR, gen_out.arguments['bootstrap_id'])
+    fetch_packages(ARTIFACT_DIR, SERVE_DIR)
+    pkgpanda.util.write_json(os.path.join(GENCONF_DIR, 'cluster_packages.json'), gen_out.cluster_packages)
```

Comment:
I could add a custom `join()` that wraps `os.path.join()` and asserts that the first arg is an absolute path. That would prevent the behavior @cmaloney mentioned above and probably let us consolidate some path assertions. I started using `os.path.join()` in the first place because I accidentally omitted a slash somewhere and got fed up.